include/aws/iot/MqttClient.h

```diff
@@ -131,16 +194,19 @@ namespace Aws
              */
             MqttClientConnectionConfig Build() noexcept;
 
-            explicit operator bool() const noexcept { return m_lastError == 0; }
-            int LastError() const noexcept { return m_lastError; }
+            explicit operator bool() const noexcept { return m_isGood; }
+            int LastError() const noexcept { return aws_last_error(); }
 
           private:
             Crt::Allocator *m_allocator;
             Crt::String m_endpoint;
             uint16_t m_portOverride;
             Crt::Io::SocketOptions m_socketOptions;
             Crt::Io::TlsContextOptions m_contextOptions;
-            int m_lastError;
+            Crt::Optional<WebsocketConfig> m_websocketConfig;
+            // this is a shared pointer for lifetime reasons.
```

Comment:
Aren't all shared_ptrs for lifetime reasons? :)

Reply:
fine....